### PR TITLE
New controller structure

### LINKED ISSUE #83: Adapt the documentation to the controller refactor
See kuzzleio/kuzzle#569 for more details.

### PR DESCRIPTION
**Pull request by dbengsch**
fix #83

### CODE REVIEW

File: source/includes/core_documentation/_01-architecture.md [hunk 1]
```diff
@@ -33,13 +33,12 @@ Focus on the above "Kuzzle kernel":
...
 
-For example, Admin, Bulk and Writer controllers emit a `data:beforeCreate` event to handle some writing actions through the storage engine.
-This event will trigger the execution of the *emit* method and of the *write* hook, which will send the received message to the internal broker.
-
-Then it is possible to implement custom hooks to trigger any event emitted by Kuzzle.
+For example, The **document** controller emits a `document:beforeCreate` event before a document is persisted in the storage engine.
 
 See also [The list of available events](#kuzzle-events-list).
 
+The event will trigger the execution of plugin functions attached to this event.
```
Comment (xbill82) [inline source/includes/core_documentation/_01-architecture.md]: "attached to this event" :point_right: "attached to it"

File: source/includes/core_documentation/_01-architecture.md [hunk 2]
```diff
@@ -33,13 +33,12 @@ Focus on the above "Kuzzle kernel":
...
+To know more, consult the [plugin documentation](#plugins).
```
Comment (xbill82) [inline source/includes/core_documentation/_01-architecture.md]: "To know more, consult" :point_right: "Learn more in"

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 3]
```diff
@@ -36,9 +36,9 @@ Sample message:
 }
 ```
 
-\#3. The ```Funnel Controller``` validates the data before sending the request to the ```Read Controller```
+\#3. The ```Funnel Controller``` validates the data before sending the request to the ```Document Controller```
 
-\#4. The ```Read Controller``` calls the ```readEngine service```
+\#4. The ```Document Controller``` calls the ```readEngine service```
 
 \#5. The ```readEngine service``` performs an HTTP Rest request to get the data from the data storage
```
Comment (xbill82) [inline source/includes/core_documentation/_02-request-scenarios.md]: Delete "Rest", by the way...

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 4]
```diff
@@ -163,16 +163,16 @@ Sample message:
...
 }
 ```
 
-\#5. The ```Funnel``` validates the message and forward the request to the ```Read Controller```
+\#5. The ```Funnel``` validates the message and forward the request to the ```Document Controller```
 
-\#6. The ```Read Controller``` calls the ```readEngine service```
+\#6. The ```Document Controller``` calls the ```readEngine service```
 
 \#7. The ```readEngine service``` performs an HTTP REST request to get the data from the data storage
```
Comment (xbill82) [inline source/includes/core_documentation/_02-request-scenarios.md]: Delete "Rest", by the way...

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 5]
```diff
@@ -376,13 +376,13 @@ Sample message:
 }
 ```
 
-\#3. The ```Funnel``` validates the message and transfer it to the ```Subscribe Controller```.
+\#3. The ```Funnel``` validates the message and transfer it to the ```Realtime Controller```.
```
Comment (xbill82) [inline source/includes/core_documentation/_02-request-scenarios.md]: "transfer" :point_right: "transfers"

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 6]
```diff
@@ -376,13 +376,13 @@ Sample message:
...
 
-\#4. The ```Subscribe Controller``` calls the ```HotelClerk``` internal component to create the subscription.
+\#4. The ```Realtime Controller``` calls the ```HotelClerk``` internal component to create the subscription.
 
 \#5. The ```HotelClerk``` calls the ```DSL``` component to get a formated filter related to the subscription (see [DSL Readme](https://github.com/kuzzleio/kuzzle/blob/master/lib/api/dsl/README.md) for more details).
 
-\#6. The ```HotelClerk``` creates a channel related to the filters and give it back to the ```Subscribe Controller```.
+\#6. The ```HotelClerk``` creates a channel related to the filters and give it back to the ```Realtime Controller```.
```
Comment (xbill82) [inline source/includes/core_documentation/_02-request-scenarios.md]: "give" :point_right: "gives"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 7]
```diff
@@ -1,248 +1,175 @@
...
+| `cleanDb:error`         | Triggered when an error occurred on clean db                     | Type: Error                                                                                   |
+
+
+### > event: collection
+
+Events triggered when a request is treated in the [`collection` controller](http://kuzzle.io/api-reference/#collection-controller).
+
+| Event                       | Description                                                  | Input         |
+|-----------------------------|--------------------------------------------------------------|---------------|
+| `collection:after<Action>`  | All actions in `collection` controller have a trigger after  | Type: Request |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "All actions in `collection` controller have a trigger after" :point_right: "All actions in the `collection` controller trigger an event after executing"
Same below :point_down:

File: source/includes/plugins/_05-kuzzle-events.md [hunk 8]
```diff
@@ -1,248 +1,175 @@
...
+|-----------------------------------------|-------------------------------------------------------------------------|----------------------|
+| `core:kuzzleStart`                      | Emitted when Kuzzle is started                                          | /                    |
+| `core:hotelClerk:addSubscription`       | Sends a diff containing the filters and internal hotelClerk updates.    | hcR object           |
+| `core:hotelClerk:join`                  | Sends hotelClerk diff when a room is joined.                            | hcR object           |
+| `core:hotelClerk:removeRoomForCustomer` | Sends the room unsubscription information if it changed.                | {connection, roomId} |
+| `core:indexCache:add`                   | Triggered if some data were actually added to Kuzzle's index cache.     | {index, collection}  |
+| `core:indexCache:remove`                | Triggered if some data were actually removed from Kuzzle's index cache. | {index, collection}  |
+| `core:indexCache:reset`                 | Triggered if the indexCache is resett.                                  | {index}              |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "resett" :point_right: "reset"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 9]
```diff
@@ -1,248 +1,175 @@
...
-|`data:beforeUpdate`   			| `write` | `update` |Triggered before controller `write` and action `update`.|Type: Request object|
+| Event                                   | Description                                                             | Input                |
+|-----------------------------------------|-------------------------------------------------------------------------|----------------------|
+| `core:kuzzleStart`                      | Emitted when Kuzzle is started                                          | /                    |
+| `core:hotelClerk:addSubscription`       | Sends a diff containing the filters and internal hotelClerk updates.    | hcR object           |
+| `core:hotelClerk:join`                  | Sends hotelClerk diff when a room is joined.                            | hcR object           |
+| `core:hotelClerk:removeRoomForCustomer` | Sends the room unsubscription information if it changed.                | {connection, roomId} |
+| `core:indexCache:add`                   | Triggered if some data were actually added to Kuzzle's index cache.     | {index, collection}  |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "Triggered if some data were actually added to Kuzzle's index cache." :question: 
Maybe "Triggered when data is added to Kuzzle's index cache."
Same below :point_down:

File: source/includes/plugins/_05-kuzzle-events.md [hunk 10]
```diff
@@ -1,248 +1,175 @@
...
+| `core:indexCache:reset`                 | Triggered if the indexCache is resett.                                  | {index}              |
+
+
+### > event: index
+
+Events triggered when a request is treated in the [`index` controller](http://kuzzle.io/api-reference/#index-controller).
+
+| Event                  | Description                                             | Input         |
+|------------------------|---------------------------------------------------------|---------------|
+| `index:after<Action>`  | All actions in `index` controller have a trigger after  | Type: Request |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: same as above (and :point_down: )

File: source/includes/plugins/_05-kuzzle-events.md [hunk 11]
```diff
@@ -1,248 +1,175 @@
...
+| `index:before<Action>` | All actions in `index` controller have a trigger before | Type: Request |
+
+
+### > event: document
+
+Events triggered when a request is treated in the [`document` controller](http://kuzzle.io/api-reference/#document-controller).
+
+| Event                     | Description                                                | Input         |
+|---------------------------|------------------------------------------------------------|---------------|
+| `document:after<Action>`  | All actions in `document` controller have a trigger after  | Type: Request |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: same as above (and :point_down: )

File: source/includes/plugins/_05-kuzzle-events.md [hunk 12]
```diff
@@ -1,248 +1,175 @@
...
 
-### > event: memoryStorage
+### > event: ms (memoryStorage)
 
-Events triggered when users manipulate data using Kuzzle memory storage capabilities.
+Events triggered when a request is done on the [`memoryStorage` controller](http://kuzzle.io/api-reference/#memorystorage-controller).
+
+| Event              | Description                                                     | Input         |
+|--------------------|-----------------------------------------------------------------|---------------|
+|`ms:after<Action>`  | All actions in `memoryStorage` controller have a trigger after  | Type: Request |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: same as above (and :point_down: )

File: source/includes/plugins/_05-kuzzle-events.md [hunk 13]
```diff
@@ -1,248 +1,175 @@
...
-|`prepareDb:updateMappingUsers`		| / | / |Triggered on Kuzzle start for creating the internal mapping for Users collection|Type: Request object.<br> Contains the default mapping in `requestObject.data.body`|
+| Event                             | Description                                                                         | Input                                                                                                   |
+|-----------------------------------|-------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------|
+| `prepareDb:createFixturesIndex`   | Triggered during database preparation. Called for each index in fixtures            | Type: Request.<br> Contains the index to create in `requestObject.index`                                |
+| `prepareDb:createInternalIndex`   | Triggered on Kuzzle start for creating the internal index `%kuzzle`                 | Type: Request.<br> Contains the internal index in `requestObject.index`                                 |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "for creating" :point_right: "to create"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 14]
```diff
@@ -1,248 +1,175 @@
...
+| `prepareDb:error`                 | Triggered when an error occurred during database preparation                        | Type: Error.                                                                                            |
+| `prepareDb:importFixtures`        | Triggered during database preparation. Called for each fixtures to import           | Type: Request.<br> Contains the index in `requestObject.index` and bulk in `requestObject.data.body`    |
+| `prepareDb:importMapping`         | Triggered during database preparation. Called for each mapping to import            | Type: Request.<br> Contains the index in `requestObject.index` and mapping in `requestObject.data.body` |
+| `prepareDb:updateMappingProfiles` | Triggered on Kuzzle start for creating the internal mapping for Profiles collection | Type: Request.<br> Contains the default mapping in `requestObject.data.body`                            |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "for creating" :point_right: "to create"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 15]
```diff
@@ -1,248 +1,175 @@
...
+| `prepareDb:updateMappingRoles`    | Triggered on Kuzzle start for creating the internal mapping for Roles collection    | Type: Request.<br> Contains the default mapping in `requestObject.data.body`                            |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "for creating" :point_right: "to create"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 16]
```diff
@@ -1,248 +1,175 @@
...
+| `prepareDb:updateMappingUsers`    | Triggered on Kuzzle start for creating the internal mapping for Users collection    | Type: Request.<br> Contains the default mapping in `requestObject.data.body`                            |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "for creating" :point_right: "to create"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 17]
```diff
@@ -1,248 +1,175 @@
...
+| Event                | Description                                                                                | Input                                                                                                                                                                          |
+|----------------------|--------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `proxy:broadcast`    | Triggered before broadcast. You can't modify the input on this event.                      | Type: Object.<br> `{payload, channelsList}`<br> `payload` is the notification content. <br>`channelsList` is an array of channels to broadcast.                                |
+| `proxy:joinChannel`  | Triggered after attach a user to a room. You can't modify the input on this event.         | Type: Object.<br> `{channel, id}`<br> `channel` is the channel name.<br> `id` is the connection id                                                                             |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "after attach" :point_right: "after attaching"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 18]
```diff
@@ -1,248 +1,175 @@
...
+| `proxy:leaveChannel` | Triggered before a room is removed for the user. You can't modify the input on this event. | Type: Object.<br> `{channel, id}`<br> `channel` is the channel name.<br> `id` is the connection id                                                                             |
+| `proxy:notify`       | Triggered before notify a connection id.                                                   | Type: Object.<br> `{payload, channelsList, id}`<br> `payload` is the notification content. <br>`channelsList` is an array of channels to notify.<br> `id` is the connection id |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "before notify" :point_right: "before notifying"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 19]
```diff
@@ -1,248 +1,175 @@
...
 
-| Event | Controller| Action | Description | Input |
-|-------|-----------|--------|-------------|-------|
-|`server:httpStarted`		| / | / |Triggered when the http server is started.|Type: String|
-|`server:overload`			| / | / |Triggered when the server overload|Type: String.<br> Contains the overload percentage with '%' character|
 
-### > event: subscription
+### > event: security
+
+Events triggered when a request is done on the [`security` controller](http://kuzzle.io/api-reference/#security-controller).
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "request is done on" :point_right: "request is sent to"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 20]
```diff
@@ -1,248 +1,175 @@
...
-### > event: room
 
-Events triggered on subscription rooms activity.
+### > event: realtime
 
-| Event | Controller| Action | Description | Input |
-|-------|-----------|--------|-------------|-------|
-|`room:new`					| / | / |Triggered when a new room is added in the rooms list. You can't modify the input on this event.|Type: Object. <br> `{roomId, index, collection, formattedFilters}`|
-|`room:remove`				| / | / |Triggered after a room is removed from the list. You can't modify the input on this event.|Type: String.<br> The room id|
+Events triggered when a request is done on the [`realtime` controller](http://kuzzle.io/api-reference/#realtime-controller).
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "request is done on" :point_right: "request is sent to"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 21]
```diff
@@ -1,248 +1,175 @@
...
+| `internalBroker:reregistering` | Triggered when the internal broker is reregistered | Type: String.<br> `'Re-registering room: ' + room`   |
+| `internalBroker:socketClosed`  | Triggered when the socket is closed                | Type: String                                         |
+| `internalBroker:started`       | Triggered when the internal broker is started      | Type: String.<br> `'Internal broker server started'` |
+
 
-### > event: memoryStorage
+### > event: ms (memoryStorage)
 
-Events triggered when users manipulate data using Kuzzle memory storage capabilities.
+Events triggered when a request is done on the [`memoryStorage` controller](http://kuzzle.io/api-reference/#memorystorage-controller).
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "request is done on" :point_right: "request is sent to"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 22]
```diff
@@ -1,248 +1,175 @@
...
-|`subscription:beforeJoin`	| `subscribe` | `join` |Triggered before controller `subscribe` and action `join`.|Type: Request object|
-|`subscription:beforeList`	| `subscribe` | `list` |Triggered before controller `subscribe` and action `list`.|Type: Request object|
-|`subscription:beforeOff`	| `subscribe` | `off` |Triggered before controller `subscribe` and action `off`.|Type: Request object|
-|`subscription:beforeOn`	| `subscribe` | `on` |Triggered before controller `subscribe` and action `on`.|Type: Request object|
-|`subscription:beforeRemoveRooms`	| `admin` | `removeRooms` |Triggered before controller `admin` and action `removeRooms`. This action remove all rooms for a given collection|Type: Request object|
 
+### > event: server
 
-### > event: workerGroup
+Events triggered on server special events or when a request is done on the [`server` controller](http://kuzzle.io/api-reference/#server-controller).
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "request is done on" :point_right: "request is sent to"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 23]
```diff
@@ -1,248 +1,175 @@
...
 
-### > event: subscription
+### > event: security
+
+Events triggered when a request is done on the [`security` controller](http://kuzzle.io/api-reference/#security-controller).
 
-Events triggered when users perform subscription related actions.
+| Event                                 | Description                                                                         | Input         |
+|---------------------------------------|-------------------------------------------------------------------------------------|---------------|
+| `security:after<Action>`              | All actions in `security` controller have a trigger after                           | Type: Request |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: same as above (and :point_down: )

File: source/includes/plugins/_05-kuzzle-events.md [hunk 24]
```diff
@@ -1,248 +1,175 @@
...
+| `security:before<Action>`             | All actions in `security` controller have a trigger before                          | Type: Request |
+| `security:formatUserForSerialization` | Triggered before serialize a user. Useful to clean a user like attribute `password` | Type: User    |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "before serialize" :point_right: "before serializing"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 25]
```diff
@@ -1,248 +1,175 @@
...
 
+### > event: server
 
-### > event: workerGroup
+Events triggered on server special events or when a request is done on the [`server` controller](http://kuzzle.io/api-reference/#server-controller).
 
-<aside class="warning">Internal use only</aside>
+| Event                   | Description                                              | Input                                                                 |
+|-------------------------|----------------------------------------------------------|-----------------------------------------------------------------------|
+| `server:after<Action>`  | All actions in `server` controller have a trigger after  | Type: Request                                                         |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: same as above (and :point_down: )

File: source/includes/plugins/_05-kuzzle-events.md [hunk 26]
```diff
@@ -1,248 +1,175 @@
...
 
+
 ### > event: auth
 
-Events triggered when users perform authentication related actions.
+Events triggered when a request is treated in the [`auth` controller](http://kuzzle.io/api-reference/#auth-controller).
+
+| Event                 | Description                                            | Input         |
+|-----------------------|--------------------------------------------------------|---------------|
+| `auth:after<Action>`  | All actions in `auth` controller have a trigger after  | Type: Request |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "All actions in `auth` controller have a trigger after" :question: 
Maybe "All actions in `auth` controller trigger an event after executing"
(same :point_down: )

File: source/includes/plugins/_05-kuzzle-events.md [hunk 27]
```diff
@@ -1,248 +1,175 @@
...
+| `auth:before<Action>` | All actions in `auth` controller have a trigger before | Type: Request |
+
+
+### > event: bulk
+
+Events triggered when a request is treated in the [`bulk` controller](http://kuzzle.io/api-reference/#bulk-controller).
+
+| Event               | Description                                        | Input         |
+|---------------------|----------------------------------------------------|---------------|
+| `bulk:afterImport`  | Triggered before the processing of action `import` | Type: Request |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: Same as above (and :point_down: )

File: source/includes/plugins/_05-kuzzle-events.md (hunk 18 above)
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: "before notify" :point_right: "before notifying"

File: source/includes/plugins/_05-kuzzle-events.md [hunk 28]
```diff
@@ -1,248 +1,175 @@
...
-| Event | Controller| Action | Description | Input |
-|-------|-----------|--------|-------------|-------|
-|`room:new`					| / | / |Triggered when a new room is added in the rooms list. You can't modify the input on this event.|Type: Object. <br> `{roomId, index, collection, formattedFilters}`|
-|`room:remove`				| / | / |Triggered after a room is removed from the list. You can't modify the input on this event.|Type: String.<br> The room id|
+Events triggered when a request is done on the [`realtime` controller](http://kuzzle.io/api-reference/#realtime-controller).
 
-### > event: security
+| Event                    | Description                                                | Input         |
+|--------------------------|------------------------------------------------------------|---------------|
+|`realtime:after<Action>`  | All actions in `realtime` controller have a trigger after  | Type: Request |
```
Comment (xbill82) [inline source/includes/plugins/_05-kuzzle-events.md]: same as above (and :point_down: )

File: source/includes/plugins/_05-kuzzle-events.md (hunk 21 above)
Comment (dbengsch) [inline source/includes/plugins/_05-kuzzle-events.md]: why here and not above or bellow ?

**Comment by dbengsch** [comment]
@xbill82 Done

**Comment by dbengsch** [comment]
Removed REST references

**Pull request merged**